packages/core/src/lib/filter/index.ts

```diff
@@ -150,53 +204,105 @@ export class FilterCore extends BaseProtocol implements IBaseProtocolCore {
     );
 
     if (!res || !res.length) {
-      throw Error(
-        `No response received for request ${request.requestId}: ${res}`
-      );
+      return {
+        failure: {
+          error: ProtocolError.REMOTE_PEER_FAULT,
+          peerId: peer.id
+        },
+        success: null
+      };
     }
 
     const { statusCode, requestId, statusDesc } =
       FilterSubscribeResponse.decode(res[0].slice());
 
     if (statusCode < 200 || statusCode >= 300) {
-      throw new Error(
+      log.error(
         `Filter unsubscribe all request ${requestId} failed with status code ${statusCode}: ${statusDesc}`
       );
+      return {
+        failure: {
+          error: ProtocolError.REMOTE_PEER_REJECTED,
```

Comment:
here and in similar places - can we map `statusCode` to some tangible error

like in case if full node has no capacity (ddos etc) - it might return some particular code 

to cover this I think we should do follow up, and sync with @NagyZoltanPeter of what error codes are expected here and in similar places

Reply:
agree - that would be useful.

we do follow generic error handling based on `statusCode`:
```ts
 const { statusCode, requestId, statusDesc } =
      FilterSubscribeResponse.decode(res[0].slice());

    if (statusCode < 200 || statusCode >= 300) {
      log.error(
        `Filter unsubscribe all request ${requestId} failed with status code ${statusCode}: ${statusDesc}`
      );
      return {
        failure: {
          error: ProtocolError.REMOTE_PEER_REJECTED,
          peerId: peer.id
        },
        success: null
      };
    }
```

creating an issue to track this: https://github.com/waku-org/js-waku/issues/1999